clang/test/CodeGenSYCL/pointers.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h %s
+// RUN: FileCheck -input-file=%t.h %s
+
+//==--pointer-int-header.cpp - USM kernel param aspace and int header test --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+template <typename name, typename Func>
```

Comment:
You can actually use functions from Inputs/sycl.hpp file, or at least fix formatting.
```suggestion
template <typename Name, typename Func>
```